addons/assigngear/functions/fnc_assignGear.sqf

```diff
@@ -106,42 +106,54 @@ _unit setUnitLoadout (configFile >> 'EmptyLoadout');
                 private _weapon = selectRandom _x;
                 if !(_weapon isEqualTo '') then {_unit addWeapon _weapon};
             };
-            case 14: { // scope
+            case 14: { // primarymagazine
+                private _magazine = selectRandom _x;
+                if !(_magazine isEqualTo '') then {_unit addPrimaryWeaponItem _magazine};
```

Comment:
With this piece of code, the open question is that if the magazine is blank should the unit spawn with or without a magazine. The common use case indicates that units should always spawn with a loaded weapon, and it would only be an edge case in mission design that a unit would spawn with a weapon but no ammunition. (E.g. a MAAWS with no ammunition to be resupplied later.)

I think if the magazine is blank, the unit should spawn with no magazine.

This preserves the current status quo of the system (since the unit can pull a magazine from their backpack to reload) and also allows mission creators to spawn units with empty weapons.

```suggestion
                private _magazine = selectRandom _x;
                if (_magazine isEqualTo '') then {
                    _unit removePrimaryWeaponItem _magazine
                } else {
                    _unit addPrimaryWeaponItem _magazine
                };
```

Need to test if ^ works.

Reply:
Pretty sure this will delete the already loaded magazine. Same for secondary and sidearm magazines, as the weapon gets loaded when it is added.

Reply:
> Pretty sure this will delete the already loaded magazine.

Yes, it will, I'm going to reorder the `loadoutArray` so that magazines are added after weapons. Currently if magazines are added first, the call to `addWeapon` will pull a magazine from inventory and add it to the weapon. From [Wiki](https://community.bistudio.com/wiki/addWeapon):

> a magazine must be added first for the added weapon to be loaded.